commit b05ae79ee3bebef9790c97eedc033d1ffb3ec39a
Author: Michel Dänzer <michel.daenzer@amd.com>
Date:   Tue Feb 23 17:19:02 2016 +0900

    glamor: Factor out glamor_set_color_depth from glamor_set_color
    
    The former takes explicit screen and depth parameters instead of
    deriving them from a pixmap.
    
    Reviewed-by: Adam Jackson <ajax@redhat.com>
    Signed-off-by: Michel Dänzer <michel.daenzer@amd.com>

--- a/glamor/glamor_transform.c
+++ b/glamor/glamor_transform.c
@@ -107,15 +107,16 @@ glamor_set_destination_drawable(Drawable
  */
 
 void
-glamor_set_color(PixmapPtr      pixmap,
-                 CARD32         pixel,
-                 GLint          uniform)
+glamor_set_color_depth(ScreenPtr      pScreen,
+                       int            depth,
+                       CARD32         pixel,
+                       GLint          uniform)
 {
     float       color[4];
 
     glamor_get_rgba_from_pixel(pixel,
                                &color[0], &color[1], &color[2], &color[3],
-                               format_for_pixmap(pixmap));
+                               format_for_depth(depth));
 
     glUniform4fv(uniform, 1, color);
 }
--- a/glamor/glamor_transform.h
+++ b/glamor/glamor_transform.h
@@ -34,9 +34,19 @@ glamor_set_destination_drawable(Drawable
                                 int             *p_off_y);
 
 void
+glamor_set_color_depth(ScreenPtr      pScreen,
+                       int            depth,
+                       CARD32         pixel,
+                       GLint          uniform);
+
+static inline void
 glamor_set_color(PixmapPtr      pixmap,
                  CARD32         pixel,
-                 GLint          uniform);
+                 GLint          uniform)
+{
+    glamor_set_color_depth(pixmap->drawable.pScreen,
+                           pixmap->drawable.depth, pixel, uniform);
+}
 
 Bool
 glamor_set_texture_pixmap(PixmapPtr    texture);
